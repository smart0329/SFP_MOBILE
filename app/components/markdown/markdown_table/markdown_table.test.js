// Copyright (c) 2019-present Smart.Inc. All Rights Reserved.
// See LICENSE.txt for license information.

import React from 'react';
import {shallowWithIntl} from 'test/intl-test-helper';

import Preferences from 'mattermost-redux/constants/preferences';

import MarkdownTable from './markdown_table';

describe('MarkdownTable', () => {
    const createCell = (type, children = null) => {
        return React.createElement('', {key: Date.now(), className: type}, children);
    };

    const numColumns = 10;
    const children = [];
    for (let i = 0; i <= numColumns; i++) {
        const cols = [];
        for (let j = 0; j <= numColumns; j++) {
            cols.push(createCell('col'));
        }

        children.push(createCell('row', cols));
    }

    const baseProps = {
        children,
        numColumns,
        theme: Preferences.THEMES.default,
    };

    test('should match snapshot', () => {
        const wrapper = shallowWithIntl(
            <MarkdownTable {...baseProps}/>,
        );

        expect(wrapper.getElement()).toMatchSnapshot();
    });

    test('should call setMaxPreviewColumns on mount', () => {
        const wrapper = shallowWithIntl(
            <MarkdownTable {...baseProps}/>,
        );
        const instance = wrapper.instance();
        const setMaxPreviewColumns = jest.spyOn(instance, 'setMaxPreviewColumns');

        instance.componentDidMount();
        expect(setMaxPreviewColumns).toHaveBeenCalled();
        expect(instance.state.maxPreviewColumns).toBeDefined();
    });

    test('should slice rows and columns', () => {
        const wrapper = shallowWithIntl(
            <MarkdownTable {...baseProps}/>,
        );

        const {maxPreviewColumns} = wrapper.state();
        expect(wrapper.find('.row')).toHaveLength(maxPreviewColumns);
        expect(wrapper.find('.col')).toHaveLength(Math.pow(maxPreviewColumns, 2));

        const newMaxPreviewColumns = maxPreviewColumns - 1;
        wrapper.setState({maxPreviewColumns: newMaxPreviewColumns});
        expect(wrapper.find('.row')).toHaveLength(newMaxPreviewColumns);
        expect(wrapper.find('.col')).toHaveLength(Math.pow(newMaxPreviewColumns, 2));
    });

    test('should add the isFirstRow prop to the first row', () => {
        const wrapper = shallowWithIntl(
            <MarkdownTable {...baseProps}/>,
        );
        const instance = wrapper.instance();
        const fullRows = instance.renderRows();
        const previewRows = instance.renderPreviewRows();

        [fullRows, previewRows].forEach((rows) => {
            const firstRows = rows.props.children.filter((child) => child.props.isFirstRow);
            expect(firstRows.length).toEqual(1);
            expect(firstRows[0]).toEqual(rows.props.children[0]);
        });
    });

    test('should add the isLastRow prop to the last row', () => {
        const wrapper = shallowWithIntl(
            <MarkdownTable {...baseProps}/>,
        );
        const instance = wrapper.instance();
        const fullRows = instance.renderRows();
        const previewRows = instance.renderPreviewRows();

        [fullRows, previewRows].forEach((rows) => {
            const lastRows = rows.props.children.filter((child) => child.props.isLastRow);
            expect(lastRows.length).toEqual(1);
            expect(lastRows[0]).toEqual(rows.props.children[rows.props.children.length - 1]);
        });
    });
});
