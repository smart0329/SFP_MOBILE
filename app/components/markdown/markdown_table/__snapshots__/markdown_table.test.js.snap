// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MarkdownTable should match snapshot 1`] = `
<TouchableWithFeedbackIOS
  onPress={[Function]}
  style={
    Object {
      "paddingRight": 10,
    }
  }
  type="opacity"
>
  <ScrollView
    onContentSizeChange={[Function]}
    onLayout={[Function]}
    scrollEnabled={false}
    showsVerticalScrollIndicator={false}
    style={
      Object {
        "maxHeight": 300,
      }
    }
  >
    <View
      style={
        Array [
          Object {
            "borderColor": "rgba(61,60,64,0.2)",
            "borderWidth": 1,
            "width": "100%",
          },
          Object {
            "width": 672,
          },
        ]
      }
    >
      <
        className="row"
        isFirstRow={true}
      >
        <
          className="col"
        />
        <
          className="col"
        />
        <
          className="col"
        />
        <
          className="col"
        />
        <
          className="col"
        />
        <
          className="col"
        />
        <
          className="col"
        />
      </>
      <
        className="row"
      >
        <
          className="col"
        />
        <
          className="col"
        />
        <
          className="col"
        />
        <
          className="col"
        />
        <
          className="col"
        />
        <
          className="col"
        />
        <
          className="col"
        />
      </>
      <
        className="row"
      >
        <
          className="col"
        />
        <
          className="col"
        />
        <
          className="col"
        />
        <
          className="col"
        />
        <
          className="col"
        />
        <
          className="col"
        />
        <
          className="col"
        />
      </>
      <
        className="row"
      >
        <
          className="col"
        />
        <
          className="col"
        />
        <
          className="col"
        />
        <
          className="col"
        />
        <
          className="col"
        />
        <
          className="col"
        />
        <
          className="col"
        />
      </>
      <
        className="row"
      >
        <
          className="col"
        />
        <
          className="col"
        />
        <
          className="col"
        />
        <
          className="col"
        />
        <
          className="col"
        />
        <
          className="col"
        />
        <
          className="col"
        />
      </>
      <
        className="row"
      >
        <
          className="col"
        />
        <
          className="col"
        />
        <
          className="col"
        />
        <
          className="col"
        />
        <
          className="col"
        />
        <
          className="col"
        />
        <
          className="col"
        />
      </>
      <
        className="row"
        isLastRow={true}
      >
        <
          className="col"
        />
        <
          className="col"
        />
        <
          className="col"
        />
        <
          className="col"
        />
        <
          className="col"
        />
        <
          className="col"
        />
        <
          className="col"
        />
      </>
    </View>
  </ScrollView>
  <LinearGradient
    colors={
      Array [
        "rgba(61,60,64,0)",
        "rgba(61,60,64,0.1)",
      ]
    }
    end={
      Object {
        "x": 1,
        "y": 0,
      }
    }
    start={
      Object {
        "x": 0,
        "y": 0,
      }
    }
    style={
      Array [
        Object {
          "borderColor": "rgba(61,60,64,0.2)",
          "borderRightWidth": 1,
          "maxHeight": 300,
          "position": "absolute",
          "top": 0,
          "width": 20,
        },
        Object {
          "height": 0,
          "left": -20,
        },
      ]
    }
  />
  <TouchableWithFeedbackIOS
    onPress={[Function]}
    style={
      Array [
        Object {
          "height": 34,
          "width": 34,
        },
        Object {
          "left": -20,
        },
      ]
    }
    type="opacity"
  >
    <View
      style={
        Array [
          Object {
            "alignItems": "flex-end",
            "maxWidth": "100%",
            "paddingBottom": 4,
            "paddingRight": 14,
            "paddingTop": 8,
          },
          Object {
            "width": 672,
          },
        ]
      }
    >
      <View
        style={
          Object {
            "alignItems": "center",
            "backgroundColor": "#ffffff",
            "borderColor": "rgba(61,60,64,0.2)",
            "borderRadius": 50,
            "borderWidth": 1,
            "display": "flex",
            "height": 34,
            "justifyContent": "center",
            "marginRight": -6,
            "marginTop": -32,
            "width": 34,
          }
        }
      >
        <Icon
          allowFontScaling={false}
          name="expand"
          size={12}
          style={
            Object {
              "color": "#2389d7",
              "fontSize": 13,
            }
          }
        />
      </View>
    </View>
  </TouchableWithFeedbackIOS>
</TouchableWithFeedbackIOS>
`;
